core/appeal/service.go

```diff
@@ -1537,7 +1539,45 @@ func getPolicy(a *domain.Appeal, p *domain.Provider, policiesMap map[string]map[
 	if resourceConfig == nil {
 		return nil, fmt.Errorf("%w: couldn't find %q resource type in the provider config", ErrInvalidResourceType, a.Resource.Type)
 	}
-	policyConfig := resourceConfig.Policy
+	policyConfig = *resourceConfig.Policy
+
+	appealMap, err := a.ToMap()
+	if err != nil {
+		return nil, fmt.Errorf("parsing appeal struct to map: %w", err)
+	}
+
+	var dynamicPolicyConfigData string
+	for _, pc := range p.Config.Policies {
+		if pc.When != "" {
+			v, err := evaluator.Expression(pc.When).EvaluateWithVars(map[string]interface{}{
+				"appeal": appealMap,
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			isFalsy := reflect.ValueOf(v).IsZero()
+			if isFalsy {
+				continue
+			}
+
+			dynamicPolicyConfigData = pc.Policy
+		}
+	}
+
+	if dynamicPolicyConfigData != "" {
+		var dynamicPolicyConfig domain.PolicyConfig
+		policyData := strings.Split(dynamicPolicyConfigData, "@")
```

Comment:
let's also additionally handle a policy that specified with latest version e.g.: `my-policy-name@latest`

Reply:
Yes, this is already handled on this splitting process. If there is no character @, then it will automatically treat the whole string as the policy name. This will be later specified as latest on getPolicy stage

Reply:
it'll still be breaking if we use `@latest` because we try to parse `latest` using `strconv.Atoi`. CMIIW

Reply:
@rahmatrhd Ah got it. Was thinking `latest` means not putting the version at all. Let's handle `policy@latest` as well then

Reply:
Added handler for `latest` version